From 12b66eeada9607aaac6d741981741c69d39beac1 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:07:01 -0400
Subject: [ethernet] qlcnic: Initialize dcbnl_ops before register_netdev

Message-id: <1408626422-3941-138-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88678
O-Subject: [PATCH RHEL 7.1 137/138] qlcnic: Initialize dcbnl_ops before register_netdev
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Rajesh Borundia <rajesh.borundia@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id cd1560e2b60fc2fbfadb9200c366eb59fe04f10d

o Initialization of dcbnl_ops after register netdev may result in
  dcbnl_ops not getting set before it is being accessed from open.
  So, moving it before register_netdev.

Signed-off-by: Rajesh Borundia <rajesh.borundia@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index 983e1dc..0fdbf34 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -2320,14 +2320,14 @@ qlcnic_setup_netdev(struct qlcnic_adapter *adapter, struct net_device *netdev,
  if (err)
   return err;
 
+ qlcnic_dcb_init_dcbnl_ops(adapter->dcb);
+
  err = register_netdev(netdev);
  if (err) {
   dev_err(&pdev->dev, "failed to register net device\n");
   return err;
  }
 
- qlcnic_dcb_init_dcbnl_ops(adapter->dcb);
-
  return 0;
 }
 
-- 
1.7.1